integrated_channels/sap_success_factors/exporters/course_metadata.py

```diff
@@ -0,0 +1,263 @@
+            this_batch = self.courses[location:location+self.CHUNK_PAGE_LENGTH]
+            location += self.CHUNK_PAGE_LENGTH
+            yield json.dumps({'ocnCourses': this_batch}, sort_keys=True).encode('utf-8')
+
+    @property
+    def data(self):
+        """
+        Return a transformed set of data ready to be exported.
+        """
+        return {
+            'courseID': self.transform_course_id,
+            'providerID': self.transform_provider_id,
+            'status': self.transform_status,
+            'title': self.transform_title,
+            'description': self.transform_description,
+            'thumbnailURI': self.transform_thumbnail_uri,
+            'content': self.transform_content,
+            'price': self.transform_price,
+            'schedule': self.transform_schedule,
+            'revisionNumber': self.transform_revision_number,
+        }
+
+    def transform_language_code(self, code):
+        """
+        Transform an ISO language code (for example, en-us) into a language name in the format read by SuccessFactors.
+        """
+        if code is None:
+            return 'English'
+
+        components = code.split('-')
+        language_code = components[0]
+        if len(components) == 2:
+            country_code = components[1]
+        elif len(components) == 1:
+            country_code = '_'
+        else:
+            raise ValueError('Language codes may only have up to two components. Could not parse: {}'.format(code))
+
+        if language_code not in SUCCESSFACTORS_OCN_LANGUAGE_CODES:
+            LOGGER.warning('Language "%s" is not supported by SAPSF. Sending English by default.', language_code)
+            return 'English'
+
+        language_family = SUCCESSFACTORS_OCN_LANGUAGE_CODES[language_code]
+        language_name = language_family.get(country_code, language_family['_'])
+        return language_name
+
+    def transform_course_id(self, course_run):
+        """
+        Return the transformed version of the course ID.
+        """
+        return course_run['key']
+
+    def transform_provider_id(self, course_run):  # pylint: disable=unused-argument
+        """
+        Return the transformed version of the provider ID.
+        """
+        return self.enterprise_configuration.provider_id
+
+    def transform_status(self, course_run):
+        """
+        Return the transformed version of the course status.
+        """
+        return (
+            self.STATUS_ACTIVE
+            if course_run['availability'] in [self.AVAILABILITY_CURRENT, self.AVAILABILITY_UPCOMING]
+            else self.STATUS_INACTIVE
+        )
+
+    def transform_title(self, course_run):
+        """
+        Return the transformed version of the course title, as well as the locale.
+        """
+        return [{
+            'locale': self.transform_language_code(course_run.get('content_language')),
+            'value': course_run.get('title') or ''
+        }]
```

Comment:
SAPSF is weird on a million levels, and I'm seriously glad Degreed had a cleaner API. Even moreso because I didn't have to do anything at all for SSO :) Just a straight up SAML IdP, no strings attached.